src/lib/components/EventStore/index.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../EventScheduleModal';
+
+import {store} from '../../store';
+
+// Pretend the date is active within this buffer. This is useful so we can
+// reveal the player for a bit more time.
+const bufferHours = 1;
+
+/**
+ * @fileoverview Publishes event data to Unistore.
+ */
+
+class EventStore extends HTMLElement {
+  constructor() {
+    super();
+    this.onStateChanged = this.onStateChanged.bind(this);
+
+    this._timeOffset = 0;
+    this._activeDay = null;
+    this._data = [];
+  }
+
+  /**
+   * @param {boolean=} change this should force a change
+   */
+  _update(change = false) {
+    const now = +new Date() + this._timeOffset;
+
+    // If there was a previously active day (because the user has their tab open for a long time),
```

Comment:
Blah sorry, I wasn't thinking through all of the states. Maybe we should add a comment at the top of the file to explain all of the states. Something like...

```
// If you hit the page 1hr before the day, you'll see that day
// If you refresh 1hr after the day, you'll see the next day
// If you leave the tab open it will continue to show the day until you refresh OR it's 1hr before the next day's start time
```

Reply:
Thanks. I've added this and the timer.